components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/GatewayArtifactsMgtDAO.java

```diff
@@ -615,14 +615,30 @@ private void removePublishedGatewayLabels(Connection connection, String apiId, S
         }
     }
 
+    private void removePublishedGatewayLabels(Connection connection, String apiId, String revision,
+        Set<String> labels) throws SQLException {
+
+        try (PreparedStatement preparedStatement = connection
+                .prepareStatement(SQLConstants.DELETE_GW_PUBLISHED_LABELS_BY_API_ID_REVISION_ID_DEPLOYMENT)) {
+            for (String label : labels) {
+                preparedStatement.setString(1, apiId);
+                preparedStatement.setString(2, revision);
+                preparedStatement.setString(3, label);
+                preparedStatement.addBatch();
+            }
+            preparedStatement.executeBatch();
+            connection.commit();
```

Comment:
We cannot do a connection.commit() here because its passed from the previous method. The caller should handle the connection closing.

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/11248/commits/51375bcdcf053c6364ea5d1b7ab02ba2be7345fd